### PR TITLE
add info on HGNC primary transcript to general report

### LINKED ISSUE #1580: HGNC Reference transcript
Hello Chiara
on behalf of my colleagues; “the chip-and-dale girls”, we would like to have the General Report to state which is the HGCN-primary reference transcript. This is for pinned variant and/or commented variants. Hope the description is clear enough?

### PR DESCRIPTION
**Pull request by northwestwitch**
Show primary HGNC reference transcript in general report. fix #1580 

**How to test**:
1. On a local instance of scout, master branch, pin a variant (POT1 is perfect for testing the PR)
1. Note that on general report any info regarding primary HGNC transcript is missing.
1. Switch to this branch and reload the report. The above info should now be present. 

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by
- [x] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
Hm, did not work out of the box locally for me. Probably this is just a residual effect of the update genes being down for a while; I'll test that. Unfortunately I'm blocked on stage atm as well, so will call it a day and get back to it tomorrow!

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 1]
```diff
@@ -643,17 +643,27 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
               <td>{{gene.region_annotation}}</td>
               <td>
                 {% set n_primary_transcripts = [] %}
+                {% set hgnc_prim = namespace(primary=None) %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/case_report.html]: 👀🤯😄
Elegant - meaning I had no idea one could do that! You will be the one updating this ... but I might try it somewhere else! :wink:

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 2]
```diff
@@ -643,17 +643,27 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
...
+                  {% if loop.index <= 5 or transcript.is_canonical or transcript.is_disease_associated or hgnc_prim.primary %} <!--Do not print too many transcripts, but include the canonical-->
                     <li>{{transcript.refseq_id}} - {{ (transcript.coding_sequence_name or '')|truncate(20, True) }} - {{ (transcript.protein_sequence_name or '')|url_decode }}
                     {% if transcript.is_canonical %}
                       <span class="badge badge-pill badge-secondary" title="canonical">C</span>
                     {% endif %}
                     {% if transcript.is_disease_associated %}
                       <span class="badge badge-pill badge-danger" title="disease_associated">D</span>
                     {% endif %}
+                    {% if hgnc_prim.primary %}
+                      <span class="badge badge-pill badge-info" title="primary_ref_transcript">hgnc primary:{{hgnc_prim.primary}}</span>
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/case_report.html]: :+1:

File: scout/server/blueprints/cases/templates/cases/case_report.html (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/case_report.html]: Me neither, I've found it on Google! 😁

**Comment by dnil** [comment]
Screenshot 2019-12-12 at 22 11 49

Wouldn't it be enough to show it once; ideally for the matching transcript - or else possibly with the pill badge in a different color for another transcript - sort of saying either "this is the transcript line you want to be looking at", or "this is a nice transcript, but we didn't find anything or the change means something else in the HGVS recommended one".

**Comment by northwestwitch** [comment]
Yes, this is not the intended behavior. Ill fix it tomorrow!

**Comment by northwestwitch** [comment]
The error should be fixed now!

**Comment by dnil** [comment]
Nicer on the eyes, though not exactly a dark mode.
There are some aesthetic issues though. For one, there is an issue with what might have been yesterdays other fix of bs4: the new(?)`<ul class="list-group">` items get a white background and need another solution. Perhaps there is a clash with the css-definition of `list-group` for the sidebar? 

Here's an example:
Screenshot 2019-12-17 at 19 26 22

**Comment by dnil** [comment]
..and this one also looks good now! :+1:

Screenshot 2019-12-17 at 21 58 21

**Pull request merged**